File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Amok.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.pixeldungeon.ml.R;
import com.nyrds.platform.util.StringsManager;
import com.watabou.pixeldungeon.sprites.CharSprite;
import com.watabou.pixeldungeon.ui.BuffIndicator;

public class Amok extends FlavourBuff {
	
	@Override
	public int icon() {
		return BuffIndicator.AMOK;
	}

	@Override
	public void attachVisual() {
        target.showStatus(CharSprite.NEGATIVE, StringsManager.getVar(R.string.Char_StaAmok));
	}
}

File: scripts/buffs/Amok.lua
---
--- Amok: berserk state, AI dispatch handled in Mob.add (was actors/buffs/Amok.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

local CharSprite = luajava.bindClass("com.watabou.pixeldungeon.sprites.CharSprite")

return buff.init{
    desc  = function ()
        return {
            icon          = 9, -- BuffIndicator.AMOK
            name          = "AmokBuff_Name",
            info          = "AmokBuff_Info",
        }
    end,

    attachVisual = function(self, buff)
        buff.target:showStatus(CharSprite.NEGATIVE, RPD.textById("Char_StaAmok"))
    end
}
